backend/src/main/java/org/sefglobal/core/academix/service/SubCategoryService.java

```diff
@@ -152,12 +152,21 @@ public SubCategory updateSubCategory(long id, SubCategory subCategory)
      * @throws ResourceNotFoundException if {@link SubCategory} for {@code id} doesn't exist
      */
     public boolean deleteSubCategory(long id) throws ResourceNotFoundException {
-        if (!subCategoryRepository.existsById(id)) {
+        Optional<SubCategory> subCategory = subCategoryRepository.findById(id);
+        if (!subCategory.isPresent()) {
             String msg = "Error, SubCategory with id: " + id + " cannot be deleted." +
                          " SubCategory doesn't exist.";
             log.error(msg);
             throw new ResourceNotFoundException(msg);
         }
+
+        List<Item> items = itemRepository.getAllBySubCategories(subCategory.get());
+        for (Item item: items) {
+            long itemId = item.getId();
+            if (item.getSubCategories().size()!=1){
+                itemRepository.getOne(itemId).getSubCategories().remove(subCategory.get());
+            }
+        }
```

Comment:
Do the following logical change.

```suggestion
        SubCategory subCategoryObj = subCategory.get();
        List<Item> items = itemRepository.getAllBySubCategories(subCategoryObj);
        for (Item item: items) {
            item.getSubCategories().remove(subCategoryObj);
            subCategoryObj.getItems().clear();
        }
```

 It's better to move the for loop code into the SubCategory class by creating the following method
`void removeItems(List<Item> items)`

Reply:
I don't think moving DAO logics to the model is a good idea.

Reply:
no, since this is logic is used to manipulate relationships it should be in the entity. 
Set methods are an example of that,
Eg: In SubCategory entity we have setItems method, etc.

Reply:
But with that, there will be isolated items. Is it ok to have isolated items?

Reply:
There was a small bug in the suggestion, it's fixed now.
However @Piumal1999 there won't be isolated Items(Items without a Subcategory). This logic will remove the SubCategory from the Item if an Item has multiple SubCategories or else this will remove the item from the table.

BTW after some research, I found out that we should be able to achieve the proper delete functionality without manually looping. You can change the parent of the relationship to SubCategory, right now the parent is Item. Can you please try to test this as well.
Reference: https://stackoverflow.com/questions/1082095/how-to-remove-entity-with-manytomany-relationship-in-jpa-and-corresponding-join

Reply:
Ah i think it will work. I'll test that also

Reply:
I changed the parent of relationship by changing mapping. After changing that, the delete mapping works fine. But then, we can't set the SubCategories of an item from the item's side.
And when we are updating a subcategory, we will have to send the list of items also with the payload.

Reply:
Added the changes